lib/IRGen/GenDecl.cpp

```diff
@@ -4202,21 +4219,26 @@ llvm::Constant *IRGenModule::emitProtocolConformances(bool asContiguousArray) {
 /// otherwise the descriptors are emitted as individual globals and nullptr is
 /// returned).
 llvm::Constant *IRGenModule::emitTypeMetadataRecords(bool asContiguousArray) {
-  if (RuntimeResolvableTypes.empty())
+  if (RuntimeResolvableTypes.empty()
+      && RuntimeResolvableTypes2.empty())
     return nullptr;
 
   std::string sectionName;
+  std::string section2Name;
   switch (TargetInfo.OutputObjectFormat) {
   case llvm::Triple::MachO:
     sectionName = "__TEXT, __swift5_types, regular";
+    section2Name = "__TEXT, __swift5_types2, regular";
     break;
   case llvm::Triple::ELF:
   case llvm::Triple::Wasm:
     sectionName = "swift5_type_metadata";
+    section2Name = "swift5_type_metadata_2";
     break;
   case llvm::Triple::XCOFF:
   case llvm::Triple::COFF:
     sectionName = ".sw5tymd$B";
+    section2Name = ".sw5tym2$B";
```

Comment:
This requires a change to stdlib/public/runtime/SwiftRT-COFF.cpp to ensure that the declarations are demarcated by the entry points.

Reply:
This is purely a future compatibility thing, so there's no immediate need to touch any runtime behavior on any platform, since we aren't yet in a place to start handling move-only types as runtime types at all. On platforms where ABI stability isn't a thing yet, we could even not emit this secondary list at all and have folks recompile when a future runtime supports move-only generics.